conanfile.py

```diff
@@ -135,6 +143,176 @@ def _python_installs(self):
 
         return python_installs
 
+    @staticmethod
+    def _is_repository_url(url):
+        # That will not work for ALL open-source projects, but should already get a large majority of them
+        return (url.startswith("https://github.com/") or url.startswith("https://gitlab.com/")) and "conan-center-index" not in url
```

Comment:
Maybe also add sourceforge? Not quite as popular as it once was (for reasons), but still a few. It's fine if not though. -- I'm OK with this approach, but it doesn't seem as robust as you'd want (urllib.parse might be overkill and not help much though...)

Reply:
For all our dependencies, I had only GitHub and Gitlab. And this is actually very convenient because they have similar "URL APIs", not sure this is the case with sourceforge